fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -461,6 +445,12 @@ def reduce(*_: Any) -> None:
                         # Normalize the bucket in one go
                         bucket.buffer.mul_(self.world_size_scaling)
 
+                        def cleanup() -> None:
```

Comment:
this should not be strictly needed if users
- use OSS.clip_grad_norm
- use grad accumulation with the no_sync() context
but in case one of these was not done, it would lead to gradients being both in the original buckets and in the reduced buckets, which can lead to errors. This is then arguably cleaner, reduced buckets are empty